### PR TITLE
fix negative-number engineering formatting

### PR DESCRIPTION
**Pull request by isentropic**
Tests passing now. The problem was with putting the floating point in case of negative numbers

### CODE REVIEW

**Comment by isentropic** [comment]
@timholy 
Answering [this](https://github.com/JuliaGraphics/Showoff.jl/pull/31#issuecomment-782843190), I think that now the bug with negative numbers for engineering notation is fixed.
Regarding the tests for the Grisu version of the code, is there a way to split the tests into two versions? Like it's done in the Showoff.jl main file, you've split the sources in case Ryu is present in the system to include ryu.jl file.

I have no idea how to split the test set gracefully for both Ryu and Grisu versions. The problem is that Grisu code was not without its issues hence these series of PRs. Perhaps it should just be declared obsolete and closed for further modification and testing. If new features to be added it should all go to ryu branch? In case of complaints from package mainteners, they could be asked to pin their dependency on the older Grisu only version of the code?

File: test/runtests.jl [hunk 1]
```diff
@@ -36,7 +36,7 @@ end
     @test Showoff.format_fixed_scientific(0.012345678, 4, true) == "12.346×10⁻³"
     @test Showoff.format_fixed_scientific(0.012345678, 4, false) == "1.2346×10⁻²"
     @test Showoff.format_fixed_scientific(-10.0, 4, false) == "-1.0000×10¹"
-    @test Showoff.format_fixed_scientific(-10.0, 4, false) == "-1.0000×10¹"
+    @test Showoff.format_fixed_scientific(-10.0, 4, true) == "-10.000×10⁰"
```
Comment (isentropic) [inline test/runtests.jl:39]: It was same with line 38, thats why I just modified this line in accordance with the bug you found

File: src/ryu.jl [hunk 2]
```diff
@@ -103,11 +103,12 @@ function get_engineering_string(x::AbstractFloat, precision::Integer)
     end
 
     buf = IOBuffer()
+    negative_base_compensation = base_digits[1] == '-' ? 1 : 0
```
Comment (isentropic) [inline src/ryu.jl:106]: I could not find a better solution to compensate for an extra index in case of negative numbers,

**Review by isentropic — COMMENTED** [review]
Negative number fixes

**Comment by timholy** [comment]
Awesome. I agree with splitting the tests since, as you point out, some of the Grisu results just seem wrong (which makes it weird to enforce them with a test). Thanks for making this more self-consistent!

**Pull request merged**